### ISSUE TITLE
Trying to exclude some populations (which I did before).  Error says to send it to you.

### ISSUE THREAD

**Issue by bmillerlab** [mennodejong1986/SambaR#29]
Hello, 
I have excluded populations after importing data using the genlight2sambar() function before and it worked fine.  I ran the filterdata() step first and then used the excludepop(do_exclude=c()) option to remove the populations.  This is the error I get now:

> excludepop(do_exclude=c("SBH","GOL","VH","EUR"))
WARNING: as of 28-10-2021, the flag do_exclude has been replaced by the flag 'popvector'.
The input vector to the popvector flag defines the names of populations to be either excluded or retained, depending on the flag 'retain' (default is FALSE, meaning: exclude).
Renaming do_exclude to popvector and continuing...
Resetting filters...
The flag 'retain' is set to FALSE, meaning that the specified populations will be excluded.
Excluding population: SBH.
Excluding population: GOL.
Excluding population: VH.
Excluding population: EUR.
Updating inds2 dataframe...
ERROR: unexpected number of rows of inds2 dataframe after adding filter1 column. Contact developer of SambaR.

I did try to use the popvector() function from the instructions in the warning in the error message ("WARNING: as of 28-10-2021, the flag do_exclude has been replaced by the flag 'popvector'.") but it returns:  Error in popvector(c("SBH", "GOL", "VH", "EUR")) : 
  could not find function "popvector"

To try and work around this I considered using the subset_pop() and going at the problem from the other direction, so I ran a small test that seemed to work but the output couldn't be run through findstructure()

> subset_pop(include_pops=c("PL-A", "PL-Gina"))
Subselecting populations. Note that this function should be executed after the filterdata() function, not before!
Done. Individuals which don't belong to populations defined in the include_pops argument, will be excluded from subsequent analyses.
This smaller dataset contains 41 retained individuals and 42349 polymorphic sites.
Don't run the filterdata() function, because this will partially undo the changes. To undo the changes, rerun the importdata() function.
> findstructure(Kmax=6,add_legend=TRUE,legend_pos="right",legend_cex=0.75)
Redefining order of populations as specified by pop_order flag.
Expected population names:
PL-A, PL-Gina
ERROR: vector input to pop_order argument is not the same length as vector input to popnames argument.

I can remove the populations using another tool before I put the data in SambaR but I think that means I can't really compare the filtered data with, and then without, these populations directly when I run the remaining functions in SambaR based on what it says in the manual.

I would appreciate any help you could offer about this problem.  Thank you.

**Comment by mennodejong1986** [comment]
Thanks for reporting the issue. I just uploaded a new SambaR script (1.08.txt) with some minor edits to the excludepop() function. Could you try to run this function again with the new script?

**Comment by bmillerlab** [comment]
Hello, Sorry I jut had a chance to come back to this issue and run the function again after uploading the new script.  Unfortunately I am getting the same error message:

excludepop(do_exclude=c("SBH","GOL","VH","EUR"))
WARNING: as of 28-10-2021, the flag do_exclude has been replaced by the flag 'popvector'.
The input vector to the popvector flag defines the names of populations to be either excluded or retained, depending on the flag 'retain' (default is FALSE, meaning: exclude).
Renaming do_exclude to popvector and continuing...
Resetting filters...
The flag 'retain' is set to FALSE, meaning that the specified populations will be excluded.
Excluding population: SBH.
Excluding population: GOL.
Excluding population: VH.
Excluding population: EUR.
Updating inds2 dataframe ...
ERROR: unexpected number of retained individuals in inds2 dataframe after updating ind2 dataframe. Contact developer of SambaR.


I looked more carefully and I do see that now when I run filterdata() the inds2  total is changing (increasing) every time, even though I am reducing the total number of individuals for the next steps as some are filtered out.  I'm not sure that is normal.
